src/localsocketcontroller.cpp

```diff
@@ -4,29 +4,21 @@
 
 #include "localsocketcontroller.h"
 
-#include <QDir>
 #include <QFileInfo>
-#include <QHostAddress>
-#include <QJsonArray>
 #include <QJsonDocument>
 #include <QJsonObject>
 #include <QJsonValue>
-#include <QStandardPaths>
 
 #include "errorhandler.h"
-#include "ipaddress.h"
 #include "leakdetector.h"
 #include "logger.h"
-#include "models/device.h"
-#include "models/keys.h"
-#include "models/server.h"
-#include "settingsholder.h"
 
-// How many times do we try to reconnect.
-constexpr int MAX_CONNECTION_RETRY = 10;
-
-// How long do we wait between one try and the next one.
-constexpr int CONNECTION_RETRY_TIMER_MSEC = 500;
+// When the daemon is unreachable, we will retry indefinitely using an
+// exponential backoff algorithm. The interval between retries starts at
+// the initial value, and doubles after each failed connection attempt,
+// with time between each retry clamped to the maximum value.
+constexpr int CONNECTION_RETRY_INITIAL_MSEC = 500;
+constexpr int CONNECTION_RETRY_MAXIMUM_MSEC = 16000;
```

Comment:
nit: Please add comments with human readable times here, e.g.
```suggestion
constexpr int CONNECTION_RETRY_MAXIMUM_MSEC = 16000; // 16s
```

Reply:
Also, I just realized this is 16s. Isn't that too little for an infinite retry loop?

Reply:
This means that if the daemon never starts, the steady-state behaviour is for the client to retry every 16 seconds indefinitely.

Reply:
Yeah I know, but still. Isn't that too small an interval? My experience with backoff intervals is all related to network requests though, so this is a genuine question lol in those cases it would be too small an interval.

Reply:
I don't think the interval is too small here. The local socket is a connection made entirely within the user's computer so the latency and resources needed to check if the daemon is active are quite small. Typically exponential backoffs are used when many clients are competing for a finite resource (either network bandwidth or time on a server) so that failures don't further compound problems of network congestion, but those concerns don't really apply here because the VPN client/daemon is a one-to-one relationship.

I kind of phrase it like, is it reasonable to expect the daemon to re-start itself and be ready to accept commands from the client within 16 seconds or so? And that feels like a yes.